osu.Game.Rulesets.Osu/Difficulty/Skills/Speed.cs

```diff
@@ -170,7 +174,17 @@ protected override double StrainValueAt(DifficultyHitObject current)
 
             currentRhythm = calculateRhythmBonus(current);
 
-            return currentStrain * currentRhythm;
+            double totalStrain = currentStrain * currentRhythm;
+
+            objectStrains.Add(totalStrain);
+
+            return totalStrain;
+        }
+
+        public double RelevantNoteCount()
+        {
+            double maxStrain = objectStrains.Max();
+            return objectStrains.Aggregate((total, next) => total + (1.0 / (1.0 + Math.Exp(-(next/maxStrain * 12.0 - 6.0)))));
```

Comment:
I'm referring to the fact that you count stream notes by taking combined finalized strains instead of doing that in the middle of processing, for example when the speed bonus happens

Reply:
It's done like this because the aggregation depends on `maxStrain` which is unknown during processing. I don't know if that term could be moved out of the exponential or not, thats beyond my math knowhow.

Reply:
What I meant is pretty much to use objects below `single_spacing_threshold` (and maybe with `speed bonus`?) instead of taking finals strains and guessing which ones are to be considered streams. That would also fix rhythmically-hard objects from being considered streams and potentially getting nerfed

Reply:
The change is meant to affect rhythmically-hard objects as well, not just streams. The goal is to scale accuracy more for the hardest 'speed' patterns, since the hardest pattern is what contributes to nearly all of the speed rating. You can't really expect to receive the full rewards if you cheesed the only hard section.

With that in mind, "guessing" by strain is the most straightforward way to count the number of hard objects in the map, because strain IS difficulty. Using only distance or deltatime doesn't tell you about the previous notes.
As an example, imagine a map with a bunch of easy triples and a long stream at the end. Those triples would count as relevant notes because they have low deltatime, and their distances are under the spacing threshold. If you completely cheesed the long stream, it wouldn't matter because the triples inflated the relevant note count. Here is such a [map](https://osu.ppy.sh/beatmapsets/369354#osu/809469) from the ranked pool.